<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="arm2">

  <xacro:property name="width" value=".2" />
  <xacro:property name="length" value=".6" />
  <xacro:property name="pi" value="3.1415" />

  <xacro:macro name="default_inertial" params="mass">
    <inertial>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <mass value="${mass}" />
        <inertia ixx="1.0" ixy="0.0" ixz="0.0"
           iyy="1.0" iyz="0.0"
           izz="1.0" />
    </inertial>
  </xacro:macro>

  <link name="base_link"/>

  <link name="first_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
         <box size="0.2 0.1 ${length}"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
         <box size="${length} .2 .1"/>
      </geometry>
    </collision>
    <xacro:default_inertial mass="10"/>
  </link>


  <link name="second_link">
    <visual>
      <origin xyz="0 0 ${-length/2}" rpy="0 0 0" />
      <geometry>
         <box size="0.2 0.1 ${length}"/>
      </geometry>
      <material name="red">
        <color rgba="0.8 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
         <box size=".2 .1 ${length}"/>
      </geometry>
    </collision>
    <xacro:default_inertial mass="10"/>
  </link>


  
  <joint name="joint_base_first" type="revolute">
    <parent link="base_link"/>
    <child link="first_link"/>   
    <origin xyz="0 0 ${length/2}" rpy="0 0 0" />
    <axis xyz="1 0 0" />
    <limit lower="-${pi}" upper="${pi}" effort="10" velocity="1.0"/>
  </joint>

  <joint name="joint_first_second" type="revolute">
    <parent link="first_link"/>
    <child link="second_link"/>   
    <origin xyz="0 0 ${length}" rpy="0 0 0" />
    <axis xyz="1 0 0" />
    <limit lower="-${pi}" upper="${pi}" effort="10" velocity="1.0"/>
  </joint>
</robot>
